src/main/java/edu/harvard/iq/dataverse/search/SearchIncludeFragment.java

```diff
@@ -1017,13 +1017,15 @@ public void setRootDv(boolean rootDv) {
         this.rootDv = rootDv;
     }
 
-    public boolean isDebug() {
-        return (debug && session.getUser().isSuperuser())
+    public Boolean getDebug() {
+        return (session.isDebug() && session.getUser().isSuperuser())
```

Comment:
Should all this logic go to the session? i.e. to use in other future pages it would be easier to not have to repeat the superuser check or the settings check?

Reply:
Not a bad idea. Consolidated in a4582f0 (where I also removed that undocumented :Debug database setting, as we discussed).

In ce78897 I made another change we discussed in Slack which is to remove the getters and setters from the search code and to rely on the getters and setters in the session code. In that commit I also attempted to explain why I'm using a Boolean with the viewParam. In short, I'm using null to mean "don't change the state".